bitsv/network/rates.py

```diff
@@ -98,23 +120,40 @@ def bsv_to_satoshi():
     return BSV
 
 
+def get_satoshi_to_fiat_rate(usd_per_bsv):
+    rate = Decimal(100000000) / Decimal(usd_per_bsv)
+    return rate
+
+
 class BitcoinSVRates:
-    # Would be better if this were HTTPS as rate information can be used
-    # maliciously.
-    SINGLE_RATE = 'http://bitcoinsv-rates.com/api/rates/'
+    # Will use the usd price from Bitfinex to then calculate the foreign exchange rates with another api:
+    BITFINEX_BSVUSD_ENDPOINT = "https://api.bitfinex.com/v1/pubticker/bsvusd"
+    FREEFOREXAPI_ENDPOINT = "https://www.freeforexapi.com/api/live?pairs="
 
     @classmethod
     def currency_to_satoshi(cls, currency):
-        r = requests.get(cls.SINGLE_RATE + currency)
-        if r.status_code != 200:
-            raise requests.exceptions.ConnectionError
-        rate = r.json()['value']
+        if currency == 'usd':
+            return cls.usd_to_satoshi()
+        else:
```

Comment:
Cosmetic but I would usually not bother with the else in this case and just drop the rest down, since you're returning.